src/backends/gitlab/implementation.js

```diff
@@ -0,0 +1,155 @@
+import trimStart from 'lodash/trimStart';
+import semaphore from "semaphore";
+import AuthenticationPage from "./AuthenticationPage";
+import API from "./API";
+import { fileExtension } from 'Lib/pathHelper';
+import { CURSOR_COMPATIBILITY_SYMBOL } from 'ValueObjects/Cursor';
+import { EDITORIAL_WORKFLOW } from "Constants/publishModes";
+
+const MAX_CONCURRENT_DOWNLOADS = 10;
+
+export default class GitLab {
+  constructor(config, options) {
+    this.config = config;
+    this.options = {
+      proxied: false,
+      API: null,
+      ...options,
+    };
+
+    if (config.getIn(["publish_mode"]) === EDITORIAL_WORKFLOW) {
+      throw new Error("The GitLab backend does not support the Editorial Workflow.")
+    }
+
+    if (!options.proxied && config.getIn(["backend", "repo"]) == null) {
```

Comment:
Shouldn't this be `this.options.proxied`?

Reply:
Good point - `options` should also have a default value of `{}` (or L17 should do a null check before using the spread operator).